### PR TITLE
immich-cli: 0.41.0 -> 2.0.6

### LINKED ISSUE #274787: Update request: immich-cli 0.41.0 → 2.0.4
- Package name: immich-cli
- Latest released version: 2.0.4

- Current version on the unstable channel: 0.41.0
- Current version on the stable/release channel: 0.41.0

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

[Immich CLI](https://immich.app/docs/features/command-line-interface/) has since been migrated from a [standalone repo](https://github.com/immich-app/CLI) (now archived) [into the main immich repository](https://github.com/immich-app/immich/tree/main/cli).

I'm actually not entirely sure if this was a migration, or a new app from scratch, in which case perhaps the current `image-cli` package should be marked as legacy, and a new one created?

**Notify maintainers**
@felschr 

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by pineapplehunter**
## Description of changes
Updates immich-cli to version 2.0.6
The immich cli version is different from immich repo, so version is imported from package.json.

closes #274787

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by cole-h** [comment]
@ofborg eval

File: pkgs/tools/misc/immich-cli/default.nix [hunk 1]
```diff
@@ -1,27 +1,29 @@
-{ lib
-, buildNpmPackage
-, fetchFromGitHub
-}:
-
+{ lib, buildNpmPackage, fetchFromGitHub }:
```
Comment (dotlambda) [inline pkgs/tools/misc/immich-cli/default.nix:1]: Let's not make unnecessary changes.
```suggestion
{ lib
, buildNpmPackage
, fetchFromGitHub
}:

```

File: pkgs/tools/misc/immich-cli/default.nix [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
 buildNpmPackage rec {
   pname = "immich-cli";
-  version = "0.41.0";
+  # version of immich and immich cli differes
+  version = (builtins.fromJSON (builtins.readFile "${src}/cli/package.json")).version;
```
Comment (dotlambda) [inline pkgs/tools/misc/immich-cli/default.nix:5]: This is an *import from derivation* which is not allowed in nixpkgs.

File: pkgs/tools/misc/immich-cli/default.nix [hunk 3]
```diff
@@ -1,27 +1,29 @@
...
-  npmDepsHash = "sha256-GOYWPRAzV59iaX32I42dOOEv1niLiDIPagzQ/QBBbKc=";
+  npmDepsHash = "sha256-NvU+v8MrwPK6q8RdVEHhzi5g6qRRmdTtInf7o2E5y6Y=";
+
+  postPatch = ''
+    cd cli
+  '';
 
   meta = {
-    changelog = "https://github.com/immich-app/CLI/releases/tag/${src.rev}";
+    changelog = "https://github.com/immich-app/immich/releases/tag/v${immich_version}";
```
Comment (dotlambda) [inline pkgs/tools/misc/immich-cli/default.nix:22]:
```suggestion
    changelog = "https://github.com/immich-app/immich/releases/tag/${src.rev}";
```

File: pkgs/tools/misc/immich-cli/default.nix [hunk 4]
```diff
@@ -1,27 +1,29 @@
...
-}:
-
+{ lib, buildNpmPackage, fetchFromGitHub }:
 buildNpmPackage rec {
   pname = "immich-cli";
-  version = "0.41.0";
+  # version of immich and immich cli differes
+  version = (builtins.fromJSON (builtins.readFile "${src}/cli/package.json")).version;
 
+  immich_version = "1.92.1";
```
Comment (dotlambda) [inline pkgs/tools/misc/immich-cli/default.nix:7]:
```suggestion
```

File: pkgs/tools/misc/immich-cli/default.nix [hunk 5]
```diff
@@ -1,27 +1,29 @@
...
   src = fetchFromGitHub {
     owner = "immich-app";
-    repo = "CLI";
-    rev = "v${version}";
-    hash = "sha256-BpJNssNTJZASH5VTgTNJ0ILj0XucWvyn3Y7hQdfCEGQ=";
+    repo = "immich";
+    rev = "v${immich_version}";
```
Comment (dotlambda) [inline pkgs/tools/misc/immich-cli/default.nix:11]:
```suggestion
    rev = "v1.92.1";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please ask upstream to separately tag CLI releases if they don't share a version number.

**Comment by pineapplehunter** [comment]
Thanks for the review! Updated the code.

> Please ask upstream to separately tag CLI releases if they don't share a version number.

I was not falimiar with the import from derivation case. (I kind of asumed that if it compiles, it is okay.)
Iwill see if I could ask for the cli version numbers, but for now, is it best to just hard code the version numbers?

**Comment by dotlambda** [comment]
> for now, is it best to just hard code the version numbers?

Yes, but `rev`should point to the actual commit at which the version was published on NPM rather than the most recent immich version.

File: pkgs/tools/misc/immich-cli/default.nix [hunk 6]
```diff
@@ -2,26 +2,30 @@
 , buildNpmPackage
 , fetchFromGitHub
 }:
-
```
Comment (dotlambda) [inline pkgs/tools/misc/immich-cli/default.nix:5]: Why remove this line?

File: pkgs/tools/misc/immich-cli/default.nix [hunk 7]
```diff
@@ -2,26 +2,30 @@
...
-  npmDepsHash = "sha256-GOYWPRAzV59iaX32I42dOOEv1niLiDIPagzQ/QBBbKc=";
+  npmDepsHash = "sha256-NvU+v8MrwPK6q8RdVEHhzi5g6qRRmdTtInf7o2E5y6Y=";
+
+  postPatch = ''
+    cd cli
+  '';
 
   meta = {
-    changelog = "https://github.com/immich-app/CLI/releases/tag/${src.rev}";
+    changelog = "https://github.com/immich-app/immich/releases/tag/v1.92.1";
```
Comment (dotlambda) [inline pkgs/tools/misc/immich-cli/default.nix:24]: This doesn't list the CLI changes.

**Comment by pineapplehunter** [comment]
@dotlambda Thanks again for the review!
I got the exact commit that is used when it was uploaded to npm, so this should be the same as using the version on npm.

**Review by dotlambda — APPROVED** [review]

**Pull request merged**